config.go

```diff
@@ -5,28 +5,44 @@ import (
 )
 
 const (
-	apiURLv1                       = "https://api.openai.com/v1"
+	openaiAPIURLv1                 = "https://api.openai.com/v1"
 	defaultEmptyMessagesLimit uint = 300
+
+	azureAPIPrefix         = "openai"
+	azureDeploymentsPrefix = "deployments"
+)
+
+type APIType string
+
+const (
+	APITypeOpenAI  APIType = "OPEN_AI"
+	APITypeAzure   APIType = "AZURE"
+	APITypeAzureAD APIType = "AZURE_AD"
 )
 
 // ClientConfig is a configuration of a client.
 type ClientConfig struct {
 	authToken string
+	BaseURL   string
+	OrgID     string
 
-	HTTPClient *http.Client
+	APIType    APIType
+	APIVersion string // required for APITypeAzure or APITypeAzureAD
```

Comment:
Maybe we can also pack Azure-related fields into a sub-structure inside the config? That way, it would be more straightforward during configuration:

```go
config.APIVersion = ...
config.Engine = ...
```
vs

```go
config.Azure.APIVersion = ...
config.Azure.Engine = ...
```

Reply:
If we do this, do we have to make a special openai config for openai?

Reply:
https://github.com/openai/openai-python/blob/main/openai/__init__.py#L39

Reply:
the py default the version to `2023-03-15-preview`, we do this the same ?

```
api_version = (
    "2023-03-15-preview" if api_type in ("azure", "azure_ad", "azuread") else None
)
```